flight/PiOS/posix/pios_fileout.c

```diff
@@ -0,0 +1,139 @@
+/**
+ ******************************************************************************
+ *
+ * @file       pios_fileout.c   
+ * @author     The OpenPilot Team, http://www.openpilot.org Copyright (C) 2012.
+ * @author     Tau Labs, http://taulabs.org, Copyright (C) 2014
+ * @author     dRonin, http://dRonin.org/, Copyright (C) 2016
+ * @brief      Pios COM driver that writes to a file.
+ * @see        The GNU Public License (GPL) Version 3
+ * @defgroup   PIOS_UDP UDP Functions
+ * @{
+ *
+ *****************************************************************************/
+/* 
+ * This program is free software; you can redistribute it and/or modify 
+ * it under the terms of the GNU General Public License as published by 
+ * the Free Software Foundation; either version 3 of the License, or 
+ * (at your option) any later version.
+ * 
+ * This program is distributed in the hope that it will be useful, but 
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY 
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License 
+ * for more details.
+ * 
+ * You should have received a copy of the GNU General Public License along 
+ * with this program; if not, write to the Free Software Foundation, Inc., 
+ * 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+ *
+ * Additional note on redistribution: The copyright and license notices above
+ * must be maintained in each individual source file that is a derivative work
+ * of this source file; otherwise redistribution is prohibited.
+ */
+
+
+/* Project Includes */
+#include "pios.h"
+
+#include <pios_fileout_priv.h>
+#include "pios_thread.h"
+#include <unistd.h>
+#include <sys/types.h>
+#include <errno.h>
+#include <fcntl.h>
+
+/* Provide a COM driver */
+static void PIOS_FILEOUT_ChangeBaud(uintptr_t udp_id, uint32_t baud);
+static void PIOS_FILEOUT_RegisterTxCallback(uintptr_t udp_id, pios_com_callback tx_out_cb, uintptr_t context);
+static void PIOS_FILEOUT_TxStart(uintptr_t udp_id, uint16_t tx_bytes_avail);
+
+const struct pios_com_driver pios_fileout_com_driver = {
+	.set_baud   = PIOS_FILEOUT_ChangeBaud,
+	.tx_start   = PIOS_FILEOUT_TxStart,
+	.bind_tx_cb = PIOS_FILEOUT_RegisterTxCallback,
+};
+
+
+static pios_fileout_dev * find_file_dev_by_id(uintptr_t tcp)
+{
+	return (pios_fileout_dev *) tcp;
+}
+
+/**
+ * Open file for writing
+ */
+struct pios_thread *tcpRxTaskHandle;
+int32_t PIOS_FILEOUT_Init(uintptr_t *file_id, const char *filename,
+		const char *mode)
+{
+	pios_fileout_dev *file_dev = PIOS_malloc(sizeof(pios_fileout_dev));
+
+	memset(file_dev, 0, sizeof(*file_dev));
+
+	file_dev->fh = fopen(filename, mode);
+
+	if (!file_dev->fh) {
+		perror("fopen");
+		return -1;
+	}
+
+	setbuf(file_dev->fh, NULL);
+
+	*file_id = (uintptr_t) file_dev;
+	
+	return 0;
+}
+
+
+void PIOS_FILEOUT_ChangeBaud(uintptr_t file_id, uint32_t baud)
```

Comment:
You don't need this function at all. PIOS_COM checks to see if the set_baud member is non-NULL before calling it.